### PR TITLE
Begin decoupling listing logic from model list implementation

### PR DESCRIPTION
**Pull request by RobotSail**
We currently handle all operations within the implementations of the CLI commands
themselves. This commit begins to refactor this by moving the implementation of
ilab model list to have the raw machinery exist in a seperate module named `machinery`

Signed-off-by: Oleg S <97077423+RobotSail@users.noreply.github.com>

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Review by cdoern — COMMENTED** [review]
we should probably talk about this in an open forum before we decide if its needed in a PR like this.

I think this was done as a step towards our REST API work, but if implemented properly the REST API probably won't necessitate sep machinery files as the API should almost be a wrapper of everything that currently exists. It can also be viewed as another entry point to the underlying click cmds.

**Comment by jaideepr97** [comment]
> we should probably talk about this in an open forum before we decide if its needed in a PR like this.
> 
> I think this was done as a step towards our REST API work, but if implemented properly the REST API probably won't necessitate sep machinery files as the API should almost be a wrapper of everything that currently exists. It can also be viewed as another entry point to the underlying click cmds.

@cdoern This was done as a first step towards https://github.com/instructlab/dev-docs/pull/131

**Comment by RobotSail** [comment]
Yeah @jaideepr97 is correct, this is so we can make a follow-up PR updating our logic of selecting models.

File: src/instructlab/machinery/listing.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                [
+                    actual_model_name,
+                    modification_time_readable,
+                    f"{adjusted_all_sizes:.1f} {magnitude}",
+                ]
+            )
+    return models
+
+
+def list_models(model_dirs: List[Path], list_checkpoints: bool):
```
Comment (jaideepr97) [inline src/instructlab/machinery/listing.py:92]: Can we specify the return type here?

**Review by jaideepr97 — COMMENTED** [review]
thanks @RobotSail 
is `machinery` supposed to function similar to a `utils` package then?

File: src/instructlab/machinery/listing.py (hunk 1 above)
Comment (RobotSail) [reply src/instructlab/machinery/listing.py]: Done

**Review by RobotSail — COMMENTED** [review]
@jaideepr97 I've moved the logic to exist under `utils.py`, but the way Python works causes us to encounter circular dependencies when the `is_model_gguf` and `is_model_safetensors` files are imported.

**Review by jaideepr97 — APPROVED** [review]
Thanks @RobotSail!

**Review by cdoern — APPROVED** [review]

**Pull request merged**